### ISSUE TITLE
How to use Supervisord on Heroku?

### ISSUE THREAD

**Issue by ronnielivingsince1994** [Supervisor/supervisor#1255]
I am trying to use Django-RQ with Supervisord on Heroku. I think I need to keep a service script in the /etc/systemd/system directory so that it can automatically start at system startup, but I'm not sure how this would translate to Heroku

I'm using Heroku CLI to push my other file within the /app directory.

Changes made via heroku run bash command are not permanent.

How do we go about it? Do I need to build a custom buildpack?

**Comment by ronnielivingsince1994** [comment]
I was following this [tutorial][1] originally, when I posted. Turns out, its rather simple to start Supervisord on Heroku, given you follow the proper configuration guidelines.

However I still need a little help here. My processes kept getting this error: 
`INFO exited: process-0 (exit status 2; not expected)`
`INFO gave up: django_rq-0 entered FATAL state, too many start retries too quickly`

I stumbled upon this #212, reading through the issue I tweaked my .conf file as below:
```
[program:django_rq]

command=/app/.heroku/python/bin/python3.6 manage.py rqworker
stdout_logfile = /app/django_rq.log
process_name=django_rq-%(process_num)s
numprocs=5
stopsignal=TERM
startsecs=0  <----
autostart=true
autorestart=true
```
Now, the Processes, are spawned>> they succeed >> then they exit. And this keeps repeating. They don't even process any task on the queue!

Can anyone in the community throw in some light over this?


[1]: https://www.vultr.com/docs/installing-and-configuring-supervisor-on-ubuntu-16-04

**Comment by ronnielivingsince1994** [comment]
I did figured it out, I was missing my directory settings! 
Added, `directory=/app/` in the .conf file, and the workers are now alive and stable.

I'm still not getting how to get them to run concurrently. 

`command=/app/.heroku/python/bin/python3.6 manage.py rqworker --burst` gets them to again start running in the abnormal mode, where worker processes enter the spawned and killed loop.
